symphony/bdk/core/service/datafeed/abstract_ackId_event_loop.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import time
+from abc import abstractmethod, ABC
+
+from symphony.bdk.core.auth.auth_session import AuthSession
+from symphony.bdk.core.config.model.bdk_config import BdkConfig
+from symphony.bdk.core.service.datafeed.abstract_datafeed_loop import AbstractDatafeedLoop
+from symphony.bdk.core.service.datafeed.exception import EventError
+from symphony.bdk.core.service.session.session_service import SessionService
+from symphony.bdk.gen.agent_api.datafeed_api import DatafeedApi
+
+EVENT_PROCESSING_MAX_DURATION_SECONDS = 30
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractAckIdEventLoop(AbstractDatafeedLoop, ABC):
+    """Base class for implementing datafeed types that relies on an ackId.
+    """
+
+    def __init__(self, datafeed_api: DatafeedApi, session_service: SessionService, auth_session: AuthSession,
+                 config: BdkConfig):
+        """
+
+        :param datafeed_api: DatafeedApi to request the service
+        :param session_service: the SessionService to get user session information
+        :param auth_session: the AuthSession instance used to get session and key manager tokens
+        :param config: the bot configuration
+        """
+        super().__init__(datafeed_api, session_service, auth_session, config)
+        self._ack_id = ""
+
+    async def recreate_datafeed(self):
```

Comment:
This method is specific to DatafeedLoopV2, it should be removed

Reply:
**_recreate_datafeed()_** and **_prepare_datafeed()_** are both specific to **_DatafeedLoopV2_** (and V1 too*) 
We have them here because **_abstract_ackId_event_loop_** extends **_abstract_datafeed_loop_** where both methods are declared. 
The interest of extending **_abstract_datafeed_loop_** is to reuse the same code as most of it is used for **DF1, DF2 and Datahose loops.** 

Options I can see:
Keep both methods declared and not implemented in DatahoseLoop.
Remove this inheritance and duplicate the code.
Introduce one more interface (abstract_recreative_datafeed_loop) and we move both methods there.

Reply:
Done in https://github.com/finos/symphony-bdk-python/pull/281/commits/f17597d4fee7a3c5c1f22afe7af066bb13073f49